lib/agent.js

```diff
@@ -261,6 +262,17 @@ class Agent {
             agentVersion: this.config.version,
             licensed: this.config.licensed
         }
+        if (this.launcher?.readPackage) {
+            const { modules } = this.launcher.readPackage()
+            if (module['node-red'] !== 'latest') {
+                state.nodeRedVersion = modules['node-red']
+            } else {
+                const nrPackPath = path.join(this.launcher.projectDir, 'node_modules/node-red/package.json')
+                const content = readFileSync(nrPackPath)
+                const packJSON = JSON.parse(content)
+                state.nodeRedVersion = packJSON.version
```

Comment:
While I am certain this will work almost every time, I am always nervous about reading files AND parsing JSON without a try..catch.

Since this is mostly informational type stuff, should we wrap this (or the whole section) in a try..catch)?

Reply:
If the node-red package.json doesn't parse it won't have been installed or the filesystem is broken and their is no way to carry on...